### PR TITLE
latex: Make it possible to use RefTeX completions even with LSP backend

### PR DESCRIPTION
**Pull request by real-or-random**
This PR adds a key binding `C-b` for `company-other-backend`, which can be pressed in the company dropdown to switch to the next completion backend.

The user can use this in the latex layer to get to RefTeX completions on demand, even with the LSP backend. This is useful because RefTeX tends to offer better completions than the LSP server texlab, at least for some things to be completed.

### CODE REVIEW

File: layers/+lang/latex/funcs.el [hunk 1]
```diff
@@ -23,31 +23,31 @@
...
-                      company-auctex-environments)
-           :modes LaTeX-mode)))
-     (when (configuration-layer/package-used-p 'company-reftex)
-       (spacemacs|add-company-backends
-         :backends company-reftex-labels
-         company-reftex-citations
-         :modes LaTeX-mode)))))
+  ;; Always activate auctex and reftex backends so that they're
+  ;; accessible via company-other-backend even when using lsp.
+  (when (configuration-layer/package-used-p 'company-auctex)
```
Comment (lebensterben) [inline layers/+lang/latex/funcs.el:28]: could we use a unified `backends` variable to store all backends. and adding individual backends to the list when conditions are met.

then there's no need to switch backend.

Comment (real-or-random) [reply layers/+lang/latex/funcs.el:28]: Do you mean a grouped backend (https://company-mode.github.io/manual/Backends.html#Grouped-Backends)?

Yes, I tried this locally before I ended up with this solution. I don't think a grouped backend is a good idea.

The issue is that both lsp and reftex suggest the same items (assuming they work properly), but reftex suggestions are better searchable. For example, lsp may suggest `ABC18` as a citation key with no additional info, and reftex may suggest the same citation with a much nicer text `Author One, Ben Author, Charlie Anotherautor: On The Usefulness of LSP. 2018. Bla....`. that, when you select it, still gives you `ABC18` but you have a much better fuzzy search.

So if we have a grouped backend, then we'd have every citation twice in the dropdown, which ends up being more confusing than it helps. I think the same is more or less true for references, too.

Comment (lebensterben) [reply layers/+lang/latex/funcs.el:28]: gotta file a feature request to lsp server.

Comment (lebensterben) [reply layers/+lang/latex/funcs.el:28]: actually you should file a request for everything you think reftex backend is doing better than lsp.

Comment (real-or-random) [reply layers/+lang/latex/funcs.el:28]: Yeah, I'm happy to do this. (That's on my list in fact.)

**Review by lebensterben — APPROVED** [review]

File: layers/+lang/latex/funcs.el (hunk 1 above)
Comment (real-or-random) [reply layers/+lang/latex/funcs.el:28]: > gotta file a feature request to lsp server.

Okay, I looked into this further. The problem is not the language server, but lsp-mode's sorting. Setting `lsp-completion-sort-initial-results` to nil fixes this, see https://github.com/ROCKTAKEY/lsp-latex/issues/38. 

If I had known this upfront, I probably wouldn't have created this PR here. However, this PR still adds a feature without drawbacks, so I think it's still useful to merge it.

**Comment by real-or-random** [comment]
Force-pushed to update the README. Now it no longer claims that reftex is better, but it's just a valid alternative.

**Review by smile13241324 — APPROVED** [review]

**Pull request merged**